byterun/floats.c

```diff
@@ -530,6 +650,49 @@ CAMLprim value caml_log1p_float(value f)
   return caml_copy_double(caml_log1p(Double_val(f)));
 }
 
+#ifndef HAS_C99_FLOAT_OPS
+static inline double simple_erf(double x)
+{
+  double a1 =  0.254829592, a2 = -0.284496736, a3 = 1.421413741,
+         a4 = -1.453152027, a5 =  1.061405429, p  = 0.3275911,
+         t, y;
+
+  int sign = (x >= 0) ? 1 : -1;
+  x = fabs(x);
+  t = 1.0 / (1.0 + p * x);
+  y = 1.0 - (((((a5 *t  + a4) * t) + a3) * t + a2) * t + a1) * t * exp(-x * x);
```

Comment:
I think that the code should at least contain a precise reference in comment to this book, if this is where you took it from.

But anyway, I have a larger concern here: it seems that erf(0) is not 0, which is quite annoying, since this means that there is a discontinuity near 0 (btw, is it even positive ? Otherwise that would mean that it is not even increasing). Even if many programs  do not depend on the fact that an approximation is good, many more actually depend on the continuity.

Why focusing on simplicity of the implementation rather than accuracy? If the user does not want accuracy, she can use such a simple implementation herself and including in stdlib is pointless.